### PR TITLE
Type Fixes and Deprecations

### LINKED ISSUE #298: [BUG] Deprecation Marking Missing for clear_scroll::path.scroll_id
### What is the bug?
Deprecation Marking Missing for clear_scroll::path.scroll_id

### What is the expected behavior?

The expected behavior is to include the following deprecation marking for the clear_scroll::path.scroll_id parameter:

```
deprecated: true
x-deprecation-message: A scroll id can be quite large and should be specified as part of the body
```

### PR DESCRIPTION
**Pull request by saimedhi**
### Description
This PR addresses type fixes for path parameters and adds deprecation for the scroll_id parameter in clear_scroll.

### Issues Resolved
#298 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — COMMENTED** [review]
Awesome. Add a test? Might need some work on getting scroll basics first.

File: spec/namespaces/cluster.yaml [hunk 1]
```diff
@@ -868,11 +868,11 @@ components:
       in: path
       name: name
       description: |-
-        Name of the component template to retrieve.
+        Names of the component templates to retrieve.
         Wildcard (`*`) expressions are supported.
       required: true
       schema:
-        $ref: '../schemas/_common.yaml#/components/schemas/Name'
+        $ref: '../schemas/_common.yaml#/components/schemas/Names'
```
Comment (Xtansia) [inline spec/namespaces/cluster.yaml:875]: `get_component_template` doesn't actually support comma-separated lists, this is something I corrected in this commit: https://github.com/opensearch-project/opensearch-api-specification/commit/7a59510d30200b22792fb0bd5dd827b620030f89

File: spec/namespaces/indices.yaml [hunk 2]
```diff
@@ -3727,7 +3727,7 @@ components:
       description: Name of the index template to retrieve. Wildcard (*) expressions are supported.
       required: true
       schema:
-        $ref: '../schemas/_common.yaml#/components/schemas/Name'
+        $ref: '../schemas/_common.yaml#/components/schemas/Names'
       style: simple
```
Comment (Xtansia) [inline spec/namespaces/indices.yaml:3731]: `get_index_template` doesn't actually support a comma separated list, this is something I corrected here: https://github.com/opensearch-project/opensearch-api-specification/commit/7a59510d30200b22792fb0bd5dd827b620030f89

File: spec/namespaces/indices.yaml [hunk 3]
```diff
@@ -2995,7 +2995,7 @@ components:
         To target all data streams in a cluster, omit this parameter or use `*`.
       required: true
       schema:
-        $ref: '../schemas/_common.yaml#/components/schemas/IndexName'
+        $ref: '../schemas/_common.yaml#/components/schemas/IndexNames'
       style: simple
```
Comment (Xtansia) [inline spec/namespaces/indices.yaml:2999]: The multi version of the `IndexName` schema is the `Indices` schema

File: spec/schemas/_common.yaml [hunk 4]
```diff
@@ -154,6 +154,12 @@ components:
       type: string
     IndexName:
       type: string
+    IndexNames:
+      oneOf:
+        - $ref: '#/components/schemas/IndexName'
+        - type: array
+          items:
+            $ref: '#/components/schemas/IndexName'
     Field:
```
Comment (Xtansia) [inline spec/schemas/_common.yaml:163]: This duplicates the `Indices` schema

**Review by Xtansia — CHANGES_REQUESTED** [review]

File: spec/namespaces/cluster.yaml (hunk 1 above)
Comment (saimedhi) [reply spec/namespaces/cluster.yaml]: @Xtansia, thank you for the review. PHP currently allows users to send name as an array. I'll update the PHP generator to ensure this isn't a breaking change. Thanks again!

**Comment by saimedhi** [comment]
I'll enhance test coverage in my next PR. Thanks!

File: spec/namespaces/_core.yaml [hunk 5]
```diff
@@ -3334,6 +3334,7 @@ components:
         Comma-separated list of scroll IDs to clear.
         To clear all scroll IDs, use `_all`.
       required: true
+      deprecated: true
```
Comment (Xtansia) [inline spec/namespaces/_core.yaml:3337]: Does this need to be specified here when it's already specified at the path level? https://github.com/opensearch-project/opensearch-api-specification/blob/84a31b0c4e38ad0f035e887fabbd0ab6701c39ce/spec/namespaces/_core.yaml#L880-L883

Comment (saimedhi) [reply spec/namespaces/_core.yaml:3337]: Here is a similar endpoint `scroll::path.scroll_id`. Deprecation is present in both places. 

https://github.com/opensearch-project/opensearch-api-specification/blob/84a31b0c4e38ad0f035e887fabbd0ab6701c39ce/spec/namespaces/_core.yaml#L5128

**Review by Xtansia — APPROVED** [review]

**Pull request merged**